### ISSUE TITLE
FunctionalSpec with @Rollback and assert in closure causes compilation error "The current scope already contains a variable of the name $spock_valueRecorder"

### ISSUE THREAD

**Issue by droggo** [grails/grails-core#9646]
Having simple functional spec like:
```
package error

import grails.test.mixin.integration.Integration
import grails.transaction.Rollback
import geb.spock.GebSpec

@Integration
@Rollback
class WithAssertFunctionalSpec extends GebSpec {
	void 'any spec method'(){
		when:
		List a = [1, 2, 3]

		then:
		a.each {
			assert it < 4
		}
	}
}
```
You will get compilation error:
```
:compileIntegrationTestGroovy
startup failed:
path\to\app\src\integration-test\groovy\error\WithAssertFunctionalSpec.groovy: -1: The current scope already contains a variable of the name $spock_valueRecorder
 @ line -1, column -1.
1 error
```

Error will disappear if either `@Rollback` annotation or `assert` inside of closure is not used. This is similar issue to https://github.com/spockframework/spock/issues/491

We have it on Grails 3.1.1

**Comment by droggo** [comment]
This happens also for integration spec which extends standard `spock.lang.Specification` instead of Geb

**Comment by hbogaards** [comment]
I got the same compiler error using Grails 3.1.1

Is this also an issue in the master (3.1.x) branch?

**Comment by graemerocher** [comment]
it is fixed in the 3.0.x branch and will be merged into 3.1.x and available in 3.0.15 and 3.1.2
